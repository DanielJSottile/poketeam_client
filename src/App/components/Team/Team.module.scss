@import '../../../index.scss';

.add-pokemon-button {
  width: 200px;
  font-size: 1.5rem;
  margin-left: auto;
  margin-right: auto;
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
  text-align: center;
  font-family: $body-font;
  padding: 5px;
  border-radius: 40px;
  border: 5px double $med-dark-grey;
}
