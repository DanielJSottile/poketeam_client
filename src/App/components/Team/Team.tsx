import React, { useContext, useState, FunctionComponent } from 'react';
import { PokemonTeam, PokemonSet } from '../../@types';
import GeneralContext from '../../contexts/GeneralContext';
import Button from '../Button';
import PokemonSetForm from '../PokemonSetForm';
import ExpandedTeam from './ExpandedTeam';
import styles from './Team.module.scss';
import UnexpandedTeam from './UnexpandedTeam';

export type TeamProps = {
  /** pokemon team */
  team: PokemonTeam;
  /** id as a team name */
  id: string;
  /** determines whether team is public or private */
  isPublic: boolean;
  /** pokemon sets, passed in usually from share page */
  /** @todo: find a way to eithe make this how its done, or remove */
  passedTeamSets?: PokemonSet[];
};

const Team: FunctionComponent<TeamProps> = ({
  team,
  id,
  isPublic,
  passedTeamSets,
}): JSX.Element => {
  const { userSets, publicSets, handleCreateDefaultPokemon, handleDeleteTeam } =
    useContext(GeneralContext);

  const [teamName, setTeamName] = useState({
    value: team.team_name || '',
    touched: false,
  });
  // const [favoriteTeam, setFavoriteTeam] = useState({ value: false, touched: false });
  // TOOD: add this as a real feature
  const [description, setDescription] = useState({
    value: team.description || '',
    touched: false,
  });
  const [teamExpandToggle, setTeamExpandToggle] = useState(true);
  const [deleteClicked, setDeleteClicked] = useState(false);

  const inputTeamName = (teamName: string) => {
    setTeamName({ value: teamName, touched: true });
  };

  // const setFavTeam = favoriteTeam => {
  //   setFavoriteTeam({value: favoriteTeam, touched: true});
  // };

  const setDesc = (description: string) => {
    setDescription({ value: description, touched: true });
  };

  const handleTeamToggle = () => {
    setTeamExpandToggle(!teamExpandToggle);
    setTeamName({ value: team.team_name || '', touched: false });
    setDescription({ value: team.description || '', touched: false });
  };

  const handleDeleteExpand = () => {
    setDeleteClicked(!deleteClicked);
  };

  /* ---------------- */

  /* Set Up Common Definitions to be 
  Used in Expanded/Unexpanded views */

  const pokemonSets = passedTeamSets
    ? passedTeamSets
    : isPublic
    ? publicSets
    : userSets;

  const pokemonSetsList = [
    ...new Set(pokemonSets.map((set: PokemonSet) => set.id)),
  ].map(
    (id) =>
      pokemonSets.find((set: PokemonSet) => set.id === id) || ({} as PokemonSet)
  );

  const teamSets = pokemonSetsList.filter(
    (set: PokemonSet) => set.team_id === team.id
  );

  const renderSetList = (teamSets: PokemonSet[]) => {
    const SetList = teamSets.map((set: PokemonSet) => {
      return <PokemonSetForm key={set.id} set={set} isPublic={isPublic} />;
    });

    if (SetList.length < 6 && !isPublic) {
      SetList.push(
        <Button
          buttonClass={styles['add-pokemon-button']}
          key={SetList.length}
          onClickCallback={(e) => {
            e.preventDefault();
            !!team.id && handleCreateDefaultPokemon(team.id);
            // we just need the id of the team.  this func fills out default vals.
          }}
        >
          Add Pokemon! +
        </Button>
      );
    }
    return SetList;
  };

  return (
    <>
      {teamExpandToggle ? (
        <UnexpandedTeam
          team={team}
          teamSets={teamSets}
          id={id}
          handleTeamToggle={handleTeamToggle}
        />
      ) : (
        <ExpandedTeam
          isPublic={isPublic}
          setList={renderSetList(teamSets)}
          teamSets={teamSets}
          id={id}
          handleTeamToggle={handleTeamToggle}
          team={team}
          teamName={teamName}
          inputTeamName={inputTeamName}
          description={description}
          setDesc={setDesc}
          handleDeleteTeam={handleDeleteTeam}
          deleteClicked={deleteClicked}
          handleDeleteExpand={handleDeleteExpand}
        />
      )}
    </>
  );
};

export default Team;
