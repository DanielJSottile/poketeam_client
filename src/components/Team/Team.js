import React, { Component, Fragment } from 'react';
import UserContext from '../../contexts/UserContext';
import showdownGenerate from '../../functions/generate';
import Set from '../Set/Set';


export default class Team extends Component {

  static contextType = UserContext;

  handleOnClick = () => {

  }

  renderExpandedTeam(){

    const {sets} = this.context;
    const {team} = this.context;

    const SetList = sets.map((set, i) => {
      return <Set key={i} id={set.id}/>
    });

    return (
      <section>
        <div className="team">
          <div className="team-header">
            <form className="team-form"> {/* on submit*/}
              <div className="team-title">
                <div>
                  <button>Delete Team X</button> {/* on delete*/}
                </div>
                <div className="title-name">
                  <label htmlFor="title-name">Team Name:</label>
                  <input className="title" placeholder="e.g. Cool Team" value={team.name} type="text" name="team-name" id={`team-name-${team.id}`}/>
                </div>
                <p>By {team.user_id}</p>
                <p>Created on: {team.date_created}</p>
                <div className="title-form">
                  <label htmlFor={`favorite-id-${team.id}`}>Favorite</label>
                  <input type="checkbox" id={`favorite-id-${team.id}`} name={`favorite-id-${team.id}`} value="true"/>
                  <p>Favorites: {team.favorites}</p>
                </div>
              </div>
              <div className="title-content">
                <label htmlFor="title-content">Description:</label>
                <textarea className="title-content desc" placeholder="e.g. description" type="text" name="title-content" id={`title-content-${team.id}`}>{team.desc}</textarea>
              </div>
              <button type="submit">Save Team Details</button>
            </form>
          <div className="export-team">
          <button>Fold Down</button> {/* onClick*/}
            <div>
              <a href="team.html" target="_blank">Share This Team!</a>
              <input type="text" readonly value={`[hostname]/share/${team.id}`}/>
            </div>
              <label htmlFor="edit-team">Export Team:</label>
              <textarea readonly type="text" name="export-team" id={`export-team-${team.id}`}>{showdownGenerate(team)}</textarea>
            </div>
          </div>
        </div>
        {SetList}
      </section>
    );
  };

  renderUnexpandedTeam() {

    const {team} = this.context;

    return (
      <section>
        <div className="team-closed">
          <div>
            <p>By {team.user_id}</p>
            <p>Created on: {team.date_created}</p>
          </div>
          <button><h4>{team.name}</h4></button> {/* onClick*/}
          <div className="title-form">
            
            <label htmlFor={`favorite-id-${team.id}`}>Favorite</label>
            <input type="checkbox" id={`favorite-id-${team.id}`} name={`favorite-id-${team.id}`} value="true"/>

            <p>Favorites: {team.favorites}</p>
        </div>
        </div>
      </section>
    );
  };

  render() {
    return (
      <Fragment>
        {this.handleOnClick() ? this.renderUnexpandedTeam() : this.renderExpandedTeam()} {/* or some value in state */ }
      </Fragment>
    );
  };
};
