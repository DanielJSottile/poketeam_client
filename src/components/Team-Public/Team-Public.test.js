import React from 'react';
import ReactDOM from 'react-dom';
//import { render } from '@testing-library/react';
import Team from './Team-Public';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<TeamPublic/>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Team />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});