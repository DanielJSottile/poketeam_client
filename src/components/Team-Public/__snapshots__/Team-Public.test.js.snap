// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamPublic/> renders without crashing 1`] = `
<Team
  team={
    Object {
      "date_created": "Thu May 21 2020 14:11:23 GMT-0400 (Eastern Daylight Time)",
      "date_modified": "",
      "description": "Test",
      "folder_id": 1,
      "folder_name": "test",
      "id": 1,
      "team_name": "Test",
      "user_id": 1,
      "user_name": "test",
    }
  }
>
  <section>
    <div
      className="team-closed"
      onClick={[Function]}
    >
      <div>
        <h3>
          Test
        </h3>
      </div>
      <div>
        <p>
          By 
          test
        </p>
        <p>
          Created on: 
          Thursday, May 21, 2020
        </p>
      </div>
    </div>
  </section>
</Team>
`;
